lib/galaxy/jobs/runners/local.py

```diff
@@ -49,23 +49,53 @@ def __init__(self, app, nworkers):
 
         super().__init__(app, nworkers)
 
+    def get_local_gpu_count(self):
+        bash_command = "/bin/bash -c 'nvidia-smi --query-gpu=index --format=csv'"
+        sp = subprocess.Popen(bash_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = sp.communicate()
+        command_not_found = "command not found"
+        if command_not_found.encode() not in out and command_not_found.encode() not in err:
+            data = (out.decode('utf-8').splitlines())[1:] #eliminate first element, as it is the header
+            return len(data)
+        return 0
+
     def __command_line(self, job_wrapper):
         """ """
         command_line = job_wrapper.runner_command_line
 
         # slots would be cleaner name, but don't want deployers to see examples and think it
         # is going to work with other job runners.
         slots = job_wrapper.job_destination.params.get("local_slots") or os.environ.get("GALAXY_SLOTS")
+        cuda_visible_devices = job_wrapper.job_destination.params.get("cuda_visible_devices") or os.environ.get("CUDA_VISIBLE_DEVICES")
+        local_gpus = job_wrapper.job_destination.params.get("local_gpus") or os.environ.get("LOCAL_GPUS")
         if slots:
             slots_statement = f'GALAXY_SLOTS="{int(slots)}"; export GALAXY_SLOTS; GALAXY_SLOTS_CONFIGURED="1"; export GALAXY_SLOTS_CONFIGURED;'
         else:
             slots_statement = 'GALAXY_SLOTS="1"; export GALAXY_SLOTS;'
 
+        num_local_gpus = self.get_local_gpu_count()
```

Comment:
Since most installations will not have GPUs, can you make this entire section conditional?
```suggestion
        if cuda_visible_devices or local_gpus:
           num_local_gpus = self.get_local_gpu_count()
           ...
```